src/pages/assets.tsx

```diff
@@ -0,0 +1,150 @@
+import { useState } from "react";
+import { api } from "@/utils/api";
+import AssetCard from "@/components/assets/AssetCard";
+import AssetsForm from "@/components/assets/AssetsForm";
+import Nav from "@/components/Nav";
+import { Button } from "@/components/ui/button";
+import {
+  Drawer,
+  DrawerClose,
+  DrawerContent,
+  DrawerFooter,
+  DrawerTrigger,
+} from "@/components/ui/drawer";
+import { useToast } from "@/components/ui/use-toast";
+
+export default function Assets() {
+  const [isDrawerOpen, setIsDrawerOpen] = useState(false);
+  const { toast } = useToast();
+
+  const { data: assets, refetch: refetchAssets } = api.assets.getAll.useQuery();
+  const { data: totalAssetValue } = api.assets.getTotalAssetValue.useQuery();
+
+  const addAssetMutation = api.assets.create.useMutation({
+    onSuccess: async () => {
+      await refetchAssets();
+      setIsDrawerOpen(false);
+      toast({
+        title: "Asset added",
+        description: "Your new asset has been added successfully.",
+      });
+    },
+    onError: (error) => {
+      toast({
+        title: "Error",
+        description: `Failed to add asset: ${error.message}`,
```

Comment:
Sometimes it's better to not tell the user exactly what went wrong with the request, and just log it. I think optimally, you'd say something like "An unexpected error occured" and you log the full error so only you as the developer are able to see what went wrong. Super minor thing.